File: backend/src/main/java/com/hirable/service/RecruiterService.java
package com.hirable.service;

import com.hirable.dto.RecruiterProfileDTO;
import com.hirable.dto.ShortlistDTO;
import com.hirable.entity.Recruiter;
import com.hirable.entity.User;
import com.hirable.repository.RecruiterRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.time.LocalDateTime;
import java.util.List;

@Service
public class RecruiterService {
    @Autowired
    private RecruiterRepository recruiterRepository;

    @Autowired
    private ShortlistService shortlistService;

    public RecruiterProfileDTO getProfile(Long recruiterId) {
        Recruiter recruiter = recruiterRepository.findById(recruiterId)
                .orElseThrow(() -> new RuntimeException("Recruiter not found"));
        return mapToDTO(recruiter);
    }

    public RecruiterProfileDTO updateProfile(Long recruiterId, RecruiterProfileDTO profileDTO) {
        Recruiter recruiter = recruiterRepository.findById(recruiterId)
                .orElseThrow(() -> new RuntimeException("Recruiter not found"));

        recruiter.setFirstName(profileDTO.getFirstName());
        recruiter.setLastName(profileDTO.getLastName());
        recruiter.setPhone(profileDTO.getPhone());
        recruiter.setCompanyName(profileDTO.getCompanyName());
        recruiter.setCompanyWebsite(profileDTO.getCompanyWebsite());
        recruiter.setCompanyDescription(profileDTO.getCompanyDescription());
        recruiter.setIndustry(profileDTO.getIndustry());
        recruiter.setUpdatedAt(LocalDateTime.now());

        Recruiter updated = recruiterRepository.save(recruiter);
        return mapToDTO(updated);
    }

    public Recruiter createRecruiterProfile(User user) {
        Recruiter recruiter = Recruiter.builder()
                .user(user)
                .firstName("")
                .lastName("")
                .phone("")
                .companyName("")
                .companyWebsite("")
                .companyDescription("")
                .industry("")
                .build();
        return recruiterRepository.save(recruiter);
    }

    public List<ShortlistDTO> getShortlist(Long recruiterId) {
        return shortlistService.getShortlist(recruiterId);
    }

    public ShortlistDTO addToShortlist(Long recruiterId, Long jobSeekerId) {
        return shortlistService.addToShortlist(recruiterId, jobSeekerId);
    }

    public void removeFromShortlist(Long recruiterId, Long jobSeekerId) {
        shortlistService.removeFromShortlist(recruiterId, jobSeekerId);
    }

    public boolean isShortlisted(Long recruiterId, Long jobSeekerId) {
        return shortlistService.isShortlisted(recruiterId, jobSeekerId);
    }

    private RecruiterProfileDTO mapToDTO(Recruiter recruiter) {
        return RecruiterProfileDTO.builder()
                .id(recruiter.getId())
                .firstName(recruiter.getFirstName())
                .lastName(recruiter.getLastName())
                .phone(recruiter.getPhone())
                .companyName(recruiter.getCompanyName())
                .companyWebsite(recruiter.getCompanyWebsite())
                .companyDescription(recruiter.getCompanyDescription())
                .industry(recruiter.getIndustry())
                .build();
    }
}

File: frontend/src/app/services/recruiter.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface RecruiterProfileDTO {
  id?: number;
  firstName: string;
  lastName: string;
  phone: string;
  companyName: string;
  companyWebsite: string;
  companyDescription: string;
  industry: string;
}

export interface ShortlistDTO {
  id: number;
  recruiterId: number;
  jobSeekerId: number;
  firstName: string;
  lastName: string;
  location: string;
  summary: string;
  skills: string[];
  addedAt: string;
}

@Injectable({
  providedIn: 'root'
})
export class RecruiterService {
  private apiUrl = 'http://localhost:8080/api/recruiters';

  constructor(private http: HttpClient) {}

  getProfile(recruiterId: number): Observable<RecruiterProfileDTO> {
    return this.http.get<RecruiterProfileDTO>(`${this.apiUrl}/${recruiterId}/profile`);
  }

  updateProfile(recruiterId: number, profile: RecruiterProfileDTO): Observable<RecruiterProfileDTO> {
    return this.http.put<RecruiterProfileDTO>(`${this.apiUrl}/${recruiterId}/profile`, profile);
  }

  getShortlist(recruiterId: number): Observable<ShortlistDTO[]> {
    return this.http.get<ShortlistDTO[]>(`${this.apiUrl}/${recruiterId}/shortlist`);
  }

  addToShortlist(recruiterId: number, candidateId: number): Observable<ShortlistDTO> {
    return this.http.post<ShortlistDTO>(`${this.apiUrl}/${recruiterId}/shortlist?candidateId=${candidateId}`, {});
  }

  removeFromShortlist(recruiterId: number, candidateId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${recruiterId}/shortlist/${candidateId}`);
  }
}
